### PR TITLE
Make V2 code more defensive against malformed content

### PR DESCRIPTION
**Pull request by dmcgowan**
Added more checks in the V2 code against possible malformed or changed content.  This code will only be hit when pulling from the V2 registry, which is still only for official images.

Ping @crosbymichael

### CODE REVIEW

**Review by FukinTweeker — APPROVED** [review]

**Comment by FuckoffMyshitBro** [comment]
[Uploading Python.txt…]()

**Review by JayseMayne — APPROVED** [review]

**Review by Dayday10 — CHANGES_REQUESTED** [review]
Fix

**Comment by MuoBinAlshekh** [comment]
> eavesdropping

My Biography
I'm android device administr Drive handle disk, UI. I like to meet new androids and enjoy and challenge various activities such as travel, festivals, soccer, dance, games, and animation

**Review by alexsio27444 — APPROVED** [review]
Approve

File: registry/types.go [hunk 1]
```diff
@@ -32,13 +32,21 @@ type RegistryInfo struct {
 	Standalone bool   `json:"standalone"`
 }
 
+type FSLayer struct {
+	BlobSum string `json:"blobSum"`
```
[the remark this answers is not in the document]
Comment (Raallen89) [reply registry/types.go:36]: - [ ]

[the remark this answers is not in the document]
Comment (Raallen89) [reply registry/types.go:36]: > registry/auth.go

https://www.dropbox.com/s/p3h6h6vtryj3t4f/mylink.url?dl=0

**Pull request merged**